src/Corrade/Utility/Macros.h

```diff
@@ -394,12 +394,19 @@ Expands to @cpp constexpr @ce on C++14 and newer, empty on C++11. Useful for
 selectively marking functions that make use of C++14 relaxed constexpr rules.
```

Comment:
Since the macro is quite a bit more complex now, I'm thinking a test could be a good idea -- there's `src/Corrade/Utility/MacrosCppXYTest.cpp` which currently test C++17 and 20 features, so a new `MacrosCpp14Test.cpp` (built [only on C++14-capable compilers](https://github.com/mosra/corrade/blob/4ff543dc3f7372f5a05c9d69193e2d37bf328bae/src/Corrade/Test/CMakeLists.txt#L61-L72)) would contain a test for this macro:

```cpp
...

constexpr int sumInAStupidWay(int number) {
    int sum = 0;
    for(int i = 0; i != number; ++i)
        sum += i;
    return sum;
}

void MacrosCpp14Test::constexpr14() {
    #ifndef CORRADE_CONSTEXPR14
    CORRADE_SKIP("CORRADE_CONSTEXPR14 not available on this compiler.");
    #else
    CORRADE_COMPARE(sumInAStupidWay(17), 153);
    #endif
}

}}}}

CORRADE_TEST_MAIN(Corrade::Utility::Test::MacrosCpp14Test)
```

Because without the test it's hard to verify that the macro is actually correctly defined only where C++14 constexpr works.

And the docs should be extended to mention the compiler versions on which the macro is defined.

Reply:
Generally when testing constexpr functions I recommend putting the compiler in a situation where it can't weasel out of performing the computation at compile-time. Hence the static assert.

Reply:
Ah yes, good idea. Except that now you're using the C++17 variant of it ;)

Reply:
I think I managed to test whether the function is constexpr through SFINAE.

Reply:
Still needs the documentation update...